### PR TITLE
Adding testing and improve documentation

### PR DESCRIPTION
**Pull request by nalakas**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/test/java/org/wso2/carbon/inbound/salesforce/poll/BearerTokenProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kesavany) [inline src/test/java/org/wso2/carbon/inbound/salesforce/poll/BearerTokenProviderTest.java]: Change the year in the license header.

File: src/test/java/org/wso2/carbon/inbound/salesforce/poll/LogginListnerTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*
```
Comment (kesavany) [inline src/test/java/org/wso2/carbon/inbound/salesforce/poll/LogginListnerTest.java]: Change the year in the license header.

File: src/test/resources/testng.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<!--
+  ~ Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kesavany) [inline src/test/resources/testng.xml]: Change the year in the license header.

Comment (nalakas) [reply src/test/resources/testng.xml]: fixed

File: src/test/java/org/wso2/carbon/inbound/salesforce/poll/BearerTokenProviderTest.java (hunk 1 above)
Comment (nalakas) [reply src/test/java/org/wso2/carbon/inbound/salesforce/poll/BearerTokenProviderTest.java]: fixed

File: src/test/java/org/wso2/carbon/inbound/salesforce/poll/LogginListnerTest.java (hunk 2 above)
Comment (nalakas) [reply src/test/java/org/wso2/carbon/inbound/salesforce/poll/LogginListnerTest.java]: fixed

**Pull request merged**